R/data.R

```diff
@@ -75,7 +104,17 @@ checkVariableIsBinary <- function(x)
     if (any(is.na(x)))
         x <- x[!is.na(x)]
     unique.values <- unique(x)
-    all(c(0, 1) %in% unique.values) && length(unique.values) == 2
+    all(unique.values %in% c(0, 1))
 }
 
-
+#' Checks if input variable x has variation and is not a constant 0 or 1
+#' @param x A binary variable, a vector with values either, 0, 1 or NA.
+#' @return Logical TRUE if constant with no variation seen. FALSE if binary.
+#' @noRd
+checkBinaryVariableNoVariation <- function(x)
```

Comment:
Would be a bit more efficient to not call `unique()` so many times, don't need to call it once to check if they are binary and then call it again to check they are constant.

Reply:
Fixed in https://github.com/Displayr/flipRegression/pull/24/commits/30c6c84cc624a352f1aeb3e9d966b7a0247eeddc